### PR TITLE
Create features.md

### PR DESCRIPTION
**Pull request by bamurtaugh**
Add the content from https://github.com/devcontainers/spec/blob/main/proposals/devcontainer-features.md. 

One to-do is to make the headers into links (i.e. `## <a href="#contribution-approaches" name="contribution-approaches" class="anchor"> Contribution approaches </a>`), but that could also be done in a subsequent PR.

### CODE REVIEW

**Comment by Chuxel** [comment]
Yeah I think this would be reasonable to add at this point along with a nav! We can then link to it from the devcontainer.json reference and the main spec. A few thoughts:
1. Maybe we should add `[Proposal]` to the heading with a brief statement saying something like "*This section provides information on a currently active proposal. See https://github.com/devcontainers/spec/issues/61 for input and links to other proposed improvements.* "?
1. It looks like this references the distribution proposal directly. We may want to pull that file in at the same time or adjust the link to point to the markdown.
1. We could reference the features section from the main spec I think just by adding a heading under "Other options". E.g., "Dev container 'features' are self-contained, shareable units of installation code and development container configuration. They are applied to container images as a secondary build step and can affect a number of dev container configuration settings. See <link> for more details."
1. In the "features" property in the devcontainer.json reference, we can reference the "devcontainer.json properties" section of this doc for more info at this point.

**Review by Chuxel — APPROVED** [review]

**Comment by bamurtaugh** [comment]
Thanks for the great feedback, @Chuxel!

I believe I addressed the 4 points above:
1. Added the word 'Proposal' and a sentence of context to the features doc
2. Pulled in the features distribution doc directly, also using 'Proposal' and the context sentence above
3. Added a features section under "Other Options" in the main spec, linking to the new features.md
4. Added a link to features.md to the features section of the devcontainer.json reference

I also made all the headers in the features and features distribution docs into links for easier sharing and navigation, consistent with the other pages of the site.

Please let me know if the changes are in-line with what you were thinking.

File: _implementors/features.md [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+layout: implementors
+title:  "Dev container Features reference [proposal]"
+shortTitle: "Features"
+author: Microsoft
+index: 5
+---
+
+> Note: This section provides information on a currently active proposal. See the [Features proposal in the spec repo](https://github.com/devcontainers/spec/issues/61) for input and links to other proposed improvements.
+
+Dev container 'features' are self-contained, shareable units of installation code and development container configuration.
```
Comment (Chuxel) [inline _implementors/features.md]:
```suggestion
Development container "Features" are self-contained, shareable units of installation code and development container configuration. The name comes from the idea that referencing one of them allows you to quickly and easily add more tooling, runtime, or library "features" into your development container for you or your collaborators to use.
```

Possible tweak to the description - we could back port this to the main spec too.  Gives a little context on why we called them "Features".

File: _implementors/json_reference.md [hunk 2]
```diff
@@ -54,7 +54,7 @@ The focus of `devcontainer.json` is to describe how to enrich a container for th
...
-| `features` (currently in preview) | object | An object of dev container features and related options to be added into your primary container. The specific options  that are available varies by feature, so see its documentation for additional details. For example: <br />`"features": {"github-cli": "latest"}` <br /><br /> ⚠️ Currently in preview. |
+| `features` (currently in preview) | object | An object of dev container features and related options to be added into your primary container. The specific options  that are available varies by feature, so see [its documentation](./features.md#a-hrefdevcontainer-json-properties-namedevcontainer-json-properties-classanchor-devcontainerjson-properties-a) for additional details. For example: <br />`"features": {"github-cli": "latest"}` <br /><br /> ⚠️ Currently in preview. |
```
Comment (Chuxel) [inline _implementors/json_reference.md]:
```suggestion
| `features` (proposed) | object | An object of dev container features and related options to be added into your primary container. The specific options  that are available varies by feature, so see [its documentation](./features.md#a-hrefdevcontainer-json-properties-namedevcontainer-json-properties-classanchor-devcontainerjson-properties-a) for additional details. For example: <br />`"features": {"github-cli": "latest"}` <br /><br /> ⚠️ Support is currently in preview as the [spec](./features.md) is not yet final. |

Maybe we specifically refer to this as proposed to tie the two together?
```

File: _implementors/spec.md [hunk 3]
```diff
@@ -67,6 +67,10 @@ In addition to the configuration options explained above, there are other settin
 
 A complete list of available metadata properties and their purposes can be found in the [`devcontainer.json` reference](https://aka.ms/devcontainer.json). However, we will describe the critical ones below in more detail.
 
+## <a href="#features-preview" name="features-preview" class="anchor"> Features (preview) </a>
+
+Dev container 'Features' are self-contained, shareable units of installation code and development container configuration. They are applied to container images as a secondary build step and can affect a number of dev container configuration settings. See the [features documentation](./features.md) for more details.
```
Comment (Chuxel) [inline _implementors/spec.md]:
```suggestion
Development container "Features" are self-contained, shareable units of installation code and development container configuration. The name comes from the idea that referencing one of them allows you to quickly and easily add more tooling, runtime, or library "features" into your development container for you or your collaborators to use.

They are applied to container images as a secondary build step and can affect a number of dev container configuration settings. See the [features documentation](./features.md) for more details.
```

Same tweak suggestion as above.

**Review by Chuxel — APPROVED** [review]
A few suggested tweaks.

**Comment by bamurtaugh** [comment]
Thanks for the suggestions! Opened https://github.com/devcontainers/spec/pull/69 with the Features wording suggestion you mentioned.

I'll merge these changes in as it seems this is a good baseline, and we can continue with other changes folks might have in mind in follow-up PRs.

**Pull request merged**